mkosi/__init__.py

```diff
@@ -984,7 +984,13 @@ def ioctl_partition_add(fd: int, nr: int, start: int, size: int) -> None:
 def ioctl_partition_remove(fd: int, nr: int) -> None:
     bp = blkpg_partition(pno=nr)
     ba = blkpg_ioctl_arg(op=BLKPG_DEL_PARTITION, data=ctypes.addressof(bp), datalen=ctypes.sizeof(bp))
-    fcntl.ioctl(fd, BLKPG, ba)
+    try:
+        fcntl.ioctl(fd, BLKPG, ba)
+    except OSError as e:
+        if e.errno != errno.EBUSY:
+            raise
+        else:
+            warn("Got EBUSY from the kernel while trying to remove partition, ignoring")
```

Comment:
Should we maybe retry after a short wait time, i.e. wrap this try in a loop?

Reply:
That's more hacks than I'm willing to do, let's see if this fixes the issue first

Reply:
Agreed. My thought was this would give the operation a chance to succeed. This way we just drop it, no?

Reply:
It's best effort with this patch, which I figure is OK. We really need to start using repart and systemd-dissect as a proper fix

Reply:
systemd-dissect is already available in Debian, repart will finally be in bookworm (to be released next year)

Reply:
But repart isn't in jammy, so we're stuck for the foreseeable future

Reply:
We could build it ourselves like we do for nspawn, couldn't we?

Reply:
We're going to try to get it into Jammy first, and maybe try shipping it in a PPA if that doesn't work